mw_jwt.go

```diff
@@ -100,31 +108,41 @@ func (k *JWTMiddleware) getSecretFromURL(url, kid, keyType string) ([]byte, erro
 	return nil, errors.New("No matching KID could be found")
 }
 
-func (k *JWTMiddleware) getIdentityFomToken(token *jwt.Token) (string, bool) {
-	// Try using a kid or sub header
+// Try using a kid or configured claim or sub claim
+func (k *JWTMiddleware) getIdentityFomToken(token *jwt.Token) (string, bool, error) {
 	idFound := false
 	var tykId string
-	if token.Header["kid"] != nil {
-		tykId = token.Header["kid"].(string)
-		idFound = true
+
+	// Global config overrides and sets the api def
+	if k.Spec.GlobalConfig.JWTSkipCheckKidAsId {
+		k.Spec.APIDefinition.JWTSkipCheckKidAsId = true
 	}
 
-	if !idFound && token.Claims.(jwt.MapClaims)["sub"] != nil {
-		tykId = token.Claims.(jwt.MapClaims)["sub"].(string)
-		idFound = true
+	// Check which claim is used for the id - kid or sub header
+	// If is not supposed to ignore KID - will use this as ID if not empty
+	if !k.Spec.APIDefinition.JWTSkipCheckKidAsId {
+		tykId, idFound = token.Header[KID].(string)
```

Comment:
better to do shorter `if tykId, idFound = token.Header[KID].(string); idFound {` if we don't need vars after `if`